### ISSUE TITLE
worldCopyJump only implemented for dragging

### ISSUE THREAD

**Issue by Driklyn** [Leaflet/Leaflet#2125]
It appears that the option `worldCopyJump` has only been implemented in [L.Map.Drag](https://github.com/Leaflet/Leaflet/blob/b4ed2bf2820161c2093209766c7505af2fce84b2/src/map/handler/Map.Drag.js#L31). It seems to me that **keyboard panning** and **mouse zooming** (since zooming to the left/right edges can change the most centered world) should have implementations as well.

http://jsfiddle.net/u9KmM

Update: Just noticed that **pan inertia** has not been implemented either, and in fact, inertia does not work very well at all with `worldCopyJump` set to true. It appears that it tries to keep you in the current centered world (like it's wrapping the end position) instead of freely flowing into the next / previous world. Should I open a separate issue for that as well?

**Comment by thatgirlcancode** [comment]
+1. I have worldCopyJump set to true. When I drag the map to the left or right, the map goes kind of crazy, autopanning backwards by several world copies with the inertia effect. I disabled inertia, but still happens. Anyone have a workaround?

**Comment by markgeraty** [comment]
:+1: 

Just wondering if there are any updates or plans for looking into this?

**Comment by pmast** [comment]
Just wanted to report the same issue. Had already prepared a playgorund for this:

https://playground-leaflet.rhcloud.com/fux/edit?html,output

**Comment by joshuakaling** [comment]
Can somebody confirm if this issue is fixed? It is open for more than 5 years?

**Comment by disarticulate** [comment]
This appears to still be an open bug.

panTo ignores world copy jump.

Here's how the map drag works:
https://github.com/Leaflet/Leaflet/blob/master/src/map/handler/Map.Drag.js
```
Map.mergeOptions({ ...
    worldCopyJump: false
})
export var Drag = Handler.extend({
	addHooks: function () {
                this._draggable = new Draggable(map._mapPane, map._container);
                ...
                this._draggable.on('predrag', this._onPreDragLimit, this);
		if (!this._draggable) {
                        ...
			if (map.options.worldCopyJump) {
				this._draggable.on('predrag', this._onPreDragWrap, this);
				map.on('zoomend', this._onZoomEnd, this);

				map.whenReady(this._onZoomEnd, this);
			}
        } 
        ...
	_onPreDragWrap: function () {
		// TODO refactor to be able to adjust map pane position after zoom
		var worldWidth = this._worldWidth,
		    halfWidth = Math.round(worldWidth / 2),
		    dx = this._initialWorldOffset,
		    x = this._draggable._newPos.x,
		    newX1 = (x - halfWidth + dx) % worldWidth + halfWidth - dx,
		    newX2 = (x + halfWidth + dx) % worldWidth - halfWidth - dx,
		    newX = Math.abs(newX1 + dx) < Math.abs(newX2 + dx) ? newX1 : newX2;

		this._draggable._absPos = this._draggable._newPos.clone();
		this._draggable._newPos.x = newX;
        },
```
Everything in panTo, viewTo, etc, don't appear to have access to worldCopyJump or this world update.
OK. So quick fix:

```
function updateTiles (e) {
      // get updated map bounds
      let bounds = this.map.getBounds()
     // extract new bounds by wrapping current bounds
      let newBounds = this.$leaflet.L.CRS.EPSG3857.wrapLatLngBounds(bounds)
     // fit the new bounds to the map, fire no events to start, duration 0 for animation 
     this.map.fitBounds(newBounds, {
        duration: 0,
        noMoveStart: true
      })
}
 function     moveendHandler (e) {
      map.off('moveend', this.moveendHandler) // turn off handler
      updateMap(e) // do arbitrary updates
      updateTiles(e.target) // update tiles (the magic
      map.on('moveend', this.moveendHandler)
}

map.on('moveend', this.moveendHandler)
```

There's a few notes around that reference wanting to move more functionality into the CRS. Also, the CRS only works if you know what projection is being used (EPSG3857) in this case.

There's still an ugly flicker on boundary changes, but that's essentially what happens with world copy jump anyways.

**Comment by jonkoops** [comment]
I believe we can close this issue, I seem to be able to pan with the keyboard just fine with `worldCopyJump` enabled on the latest version of Leaflet.

If I am mistaken, please let me know and I will re-open the issue.

**Comment by Malvoz** [comment]
@jonkoops There does seem to be a discrepancy in the behavior/position of the circleMarker depending on whether you pan using a mouse or a keyboard in the [provided fiddle](https://github.com/Leaflet/Leaflet/issues/2125#issue-21545793). TBH I can't tell if that's entailed in the problem description here, but let's keep this open for now.

**Comment by Falke-Design** [comment]
@Malvoz this is fixed with #8562 . Please test the fiddle with the current main: http://jsfiddle.net/1t0v67ne/

**Comment by Malvoz** [comment]
@Falke-Design Ah, I just saw https://github.com/Leaflet/Leaflet/pull/8562. Thanks.
